### PR TITLE
Branch165 callout need table

### PR DESCRIPTION
**Pull request by timwilson139**
Added some mock data for donations, needs, and needcallouts. Updated the database schema in interfaith_DDL.sql. Updated the init-user-db.sh file for Docker.

### CODE REVIEW

**Review by TylerNicholls — APPROVED** [review]
Potential mock data issues:
donation quantity > needcallout quantity
donation date > callout end_date
donation date < callout start_date

File: docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql [hunk 1]
```diff
@@ -46,13 +54,22 @@ CREATE TABLE need (
...
+  need_id BIGINT REFERENCES needs,
+  callout_id BIGINT REFERENCES callouts,
+  needcallout_active BOOLEAN NOT NULL
+);
+
+DROP TABLE IF EXISTS donations cascade;
+CREATE TABLE donations (
+  donation_id BIGSERIAL PRIMARY KEY,
+  donation_quantity INT NOT NULL,
+  donation_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
```
Comment (Tiernebre) [inline docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:70]: Is this the created date or last modified date? Either way, it is doing the correct thing so it's not a blocking issue

File: docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql [hunk 2]
```diff
@@ -46,13 +54,22 @@ CREATE TABLE need (
...
---   user_id BIGINT REFERENCES user,
---   callout_id BIGINT REFERENCES callout,
---   active BOOLEAN NOT NULL
--- );
+DROP TABLE IF EXISTS needcallouts cascade;
+CREATE TABLE needcallouts (
+  needcallout_id BIGSERIAL PRIMARY KEY,
+  needcallout_quantity INT NOT NULL,
+  need_id BIGINT REFERENCES needs,
+  callout_id BIGINT REFERENCES callouts,
```
Comment (Tiernebre) [inline docker/postgres/docker-entrypoint-initdb.d/mock-data/interfaith_DDL.sql:62]: need_id and callout_id should be not nullable

**Review by Tiernebre — APPROVED** [review]
I approve the pull request. If you get time to fix the stuff I gave feedback on that would be great, but they're not worth declining this pull request for.

**Pull request merged**